servant-server/src/Servant/Server.hs

```diff
@@ -126,8 +138,8 @@ import           Servant.Server.Internal
 -- > main :: IO ()
 -- > main = Network.Wai.Handler.Warp.run 8080 app
 --
-serve :: (HasServer api '[]) => Proxy api -> Server api -> Application
-serve p = serveWithContext p EmptyContext
+serve :: (HasServer api DefaultErrorFormatters) => Proxy api -> Server api -> Application
+serve p = serveWithContext p defaultErrorFormatters
```

Comment:
Doesn't this mean that if you want to add your own formatters, you need to define variants of `serve`, `hoistServer`, ...?

Also, what happens if the context is to be used for something else?  (I've never used servant contexts and I'm not sure how they work.)

Reply:
Yes, so this is kinda a breaking change.

If one needs to use one's custom context and wants to keep default handlers, custom context should just be cons'ed on top of the default one. An example can be found [here](https://github.com/haskell-servant/servant/pull/1312/files#diff-53827cb3f09ba7c0122b0f4349f0b86aR83):

```haskell
comprehensiveApiContext :: Context (NamedContext "foo" '[] ': DefaultErrorFormatters)
comprehensiveApiContext = NamedContext EmptyContext :. defaultErrorFormatters
```

Thanks to `HasContextEntry` picking the first entry with the correct type, a user can add custom formatter of one type without needing to override others. See example in `greet.hs` (I'm going to revert that change, it's purely for demonstration of the PR).

```haskell
test = serveWithContext testApi (customFormatter :. defaultErrorFormatters) server
```

Here the context will get two formatters of `BodyParseErrorFormatter` type, but user's one will be picked at runtime.

I could've made another typeclass `MaybeHasContextEntry` that will pick something from context if it's present and return `Nothing` otherwise, but I find my approach more explicit.